pkg/cmd/recycle/recycle.go

```diff
@@ -50,3 +57,15 @@ func Recycle(dir string) error {
 		return os.Remove(path)
 	}).Walk(dir)
 }
+
+// CheckEmpty returns error if specified directory is not empty.
+func CheckEmpty(dir string) error {
+	return newWalker(func(path string, info os.FileInfo) error {
```

Comment:
IMO, it's single NFS mount, ".snapshot" (or whatever is it called) won't have a different FSID.